File: cs/src/AutoDiff/Zero.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AutoDiff
{
    /// <summary>
    /// A constant zero term. Similar to <see cref="Constant"/> but represents only the value 0.
    /// </summary>
    [Serializable]
    public class Zero : Term
    {
		

        /// <summary>
        /// Accepts a term visitor
        /// </summary>
        /// <param name="visitor">The term visitor to accept.</param>
        public override void Accept(ITermVisitor visitor)
        {
            visitor.Visit(this);
        }

        /// <summary>
        /// Accepts a term visitor with a generic result
        /// </summary>
        /// <typeparam name="TResult">The type of the result from the visitor's function</typeparam>
        /// <param name="visitor">The visitor to accept</param>
        /// <returns>
        /// The result from the visitor's visit function.
        /// </returns>
        public override TResult Accept<TResult>(ITermVisitor<TResult> visitor)
        {
            return visitor.Visit(this);
        }
        
        public override string ToString()
		{
			return "0";
		}
		
		public override Term AggregateConstants()
		{
			return this;
		}
		public override Term Derivative(Variable v)
		{
			return this;
		}
		public override bool Equals (object obj)
		{
			if (obj is Zero) return true;
			return false;
			
		}
    }
}

File: java/src/main/java/AutoDiff/Zero.java
//
// Translated by CS2J (http://www.cs2j.com): 12.11.2016 09:40:34
//

package AutoDiff;

/**
* A constant zero term. Similar to 
*  {@link #Constant}
*  but represents only the value 0.
*/
public class Zero  extends Term 
{
//    /**
//    * Accepts a term visitor
//    *
//    *  @param visitor The term visitor to accept.
//    */
//    public void accept(ITermVisitor visitor) throws Exception {
//        visitor.visit(this);
//    }

    /**
    * Accepts a term visitor with a generic result
    * The type of the result from the visitor's function
    *  @param visitor The visitor to accept
    *  @return 
    * The result from the visitor's visit function.
    */
    public <TResult>TResult accept(ITermVisitor<TResult> visitor) throws Exception {
        return visitor.visit(this);
    }

    public String toString() {
        try
        {
            return "0";
        }
        catch (RuntimeException exception)
        {
            throw exception;
        }
        catch (Exception exception)
        {
            throw new RuntimeException(exception);
        }
    
    }

    public Term aggregateConstants() throws Exception {
        return this;
    }

    public Term derivative(Variable v) throws Exception {
        return this;
    }

    public boolean equals(Object obj) {
        try
        {
            if (obj instanceof Zero)
                return true;
             
            return false;
        }
        catch (RuntimeException dummyCatchVar1)
        {
            throw dummyCatchVar1;
        }
        catch (Exception dummyCatchVar1)
        {
            throw new RuntimeException(dummyCatchVar1);
        }
    
    }

}


